### PR TITLE
chore(pymongo): add testing to pymongo patch

### PR DESCRIPTION
**Pull request by quinna-h**
Tests patching and unpatching pymongo, following the new changes to add support for pymongo v4.9.1 in [feat(pymongo): add support for v4.9](https://github.com/DataDog/dd-trace-py/pull/10718)

Additionally, adds fix for an issue with `assert_wrapped` that previously assumed wrapped object always comes from `wrapt`, when it can come from internal wrapping function.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: tests/contrib/pymongo/test_pymongo_patch.py [hunk 1]
```diff
@@ -21,11 +21,76 @@ class TestPymongoPatch(PatchTestCase.Base):
     __unpatch_func__ = unpatch
     __get_version__ = get_version
 
+    def _get_imports(self, pymongo):
+        if _VERSION >= (4, 9):
```
Comment (mabdinur) [inline tests/contrib/pymongo/test_pymongo_patch.py:25]: Do don't need to the nest these imports in a function? We can just make them global?

File: tests/contrib/pymongo/test_pymongo_patch.py [hunk 2]
```diff
@@ -2,13 +2,13 @@
 # script. If you want to make changes to it, you should make sure that you have
 # removed the ``_generated`` suffix from the file name, to prevent the content
 # from being overwritten by future re-generations.
-
-from ddtrace.contrib.pymongo import get_version
-from ddtrace.contrib.pymongo.patch import patch
+from ddtrace.contrib.internal.pymongo.patch import _VERSION
+from ddtrace.contrib.internal.pymongo.patch import get_version
+from ddtrace.contrib.internal.pymongo.patch import patch
```
Comment (mabdinur) [inline tests/contrib/pymongo/test_pymongo_patch.py:7]:
```suggestion
from ddtrace.contrib.pymongo import patch
```

Comment (mabdinur) [inline tests/contrib/pymongo/test_pymongo_patch.py:7]:
```suggestion
from ddtrace.contrib.pymongo import get_version
from ddtrace.contrib.pymongo import patch
```

Comment (mabdinur) [reply tests/contrib/pymongo/test_pymongo_patch.py:7]: nit: We should import patch and unpatch from the public api

File: tests/contrib/pymongo/test_pymongo_patch.py [hunk 3]
```diff
@@ -2,13 +2,13 @@
...
 
 
 try:
```
Comment (mabdinur) [inline tests/contrib/pymongo/test_pymongo_patch.py:10]:
```suggestion
```

File: tests/contrib/pymongo/test_pymongo_patch.py [hunk 4]
```diff
@@ -2,13 +2,13 @@
...
-    from ddtrace.contrib.pymongo.patch import unpatch
+    from ddtrace.contrib.internal.pymongo.patch import unpatch
 except ImportError:
     unpatch = None
```
Comment (mabdinur) [inline tests/contrib/pymongo/test_pymongo_patch.py:13]:
```suggestion
```

File: tests/contrib/pymongo/test_pymongo_patch.py [hunk 5]
```diff
@@ -2,13 +2,13 @@
...
-from ddtrace.contrib.pymongo import get_version
-from ddtrace.contrib.pymongo.patch import patch
+from ddtrace.contrib.internal.pymongo.patch import _VERSION
+from ddtrace.contrib.internal.pymongo.patch import get_version
+from ddtrace.contrib.internal.pymongo.patch import patch
 
 
 try:
-    from ddtrace.contrib.pymongo.patch import unpatch
+    from ddtrace.contrib.internal.pymongo.patch import unpatch
```
Comment (mabdinur) [inline tests/contrib/pymongo/test_pymongo_patch.py:11]: I don't think we need this try-except. The unpatch function should always exist 

```suggestion
from ddtrace.contrib.pymongo import unpatch
```

**Review by mabdinur — COMMENTED** [review]
Left some nits but overall looks great!

I updated the base branch from main to munir/test-pymongo-4-9. We should remove the kafka changes from this PR

File: tests/contrib/pymongo/test_pymongo_patch.py [hunk 6]
```diff
@@ -22,10 +33,54 @@ class TestPymongoPatch(PatchTestCase.Base):
...
+            self.assert_not_wrapped(Server.checkout)
+        else:
+            self.assert_not_wrapped(Server.get_socket)
+
+        self.assert_not_wrapped(Connection.command)
+        self.assert_not_wrapped(Connection.write_command)
 
     def assert_not_module_double_patched(self, pymongo):
-        pass
+        self.assert_not_double_wrapped(Topology.select_server)
```
Comment (erikayasuda) [inline tests/contrib/pymongo/test_pymongo_patch.py:72]: Maybe a silly question, but do we not need to check whether `pymongo.MongoClient.__init__` is double wrapped too?

**Review by erikayasuda — APPROVED** [review]

File: tests/contrib/pymongo/test_pymongo_patch.py (hunk 6 above)
Comment (quinna-h) [reply tests/contrib/pymongo/test_pymongo_patch.py:72]: Good catch thank you!

**Pull request merged**